index.js

```diff
@@ -229,10 +230,17 @@ module.exports = {
     }
 
     if (ext === 'js') {
+      const scriptAttributes = [];
+
       if (this.js.useAsync) {
-        return '<script async src="' + path + '"></script>\n';
+        scriptAttributes.push(' async');
+      }
+
+      if (this.js.useDefer) {
+        scriptAttributes.push(' defer');
       }
-      return '<script src="' + path + '"></script>\n';
+  
+      return '<script' + scriptAttributes.join('') + ' src="' + path + '"></script>\n';
```

Comment:
What do you think about adding the space in the `join()`? That seems less prone to future contribution error. For example:

```js
scriptAttributes.push('async');
```

Then
```js
return '<script' + scriptAttributes.join(' ') + ' src="' + path + '"></script>\n';
```